window_manager/src/XWindow.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef RCBE_ENGINE_XWINDOW_HPP
+#define RCBE_ENGINE_XWINDOW_HPP
+
+#include <memory>
+#include <mutex>
+#include <atomic>
+#include <type_traits>
+
+#include <core/WindowConfig.hpp>
+#include <data_types/rendering/RenderingContext.hpp>
+
+namespace rcbe::core {
+class XWindow {
+public:
+
+    XWindow(WindowConfig &&config_, Display* root_display, int screen_number, const Window& root_window, const Atom& delete_msg);
+    ~XWindow();
+
+    [[nodiscard]] const WindowConfig& get_config() const;
+
+
+    decltype(auto) start_window_loop_aync();
+    void start_window_loop();
+    void stop_window_loop();
+
+    void on_configure(window::configure_handler_t&& handler);
+    void on_unmap(window::unmap_handler_t&& handler);
+
+    [[nodiscard]]const std::shared_ptr<rendering::RenderingContext>& get_context() const;
+
+    void kill();
+    void map_window();
+
+private:
+
+    void window_loop();
+
+    WindowConfig config_;
+    std::mutex running_mutex_;
+    bool running_ = false;
+    Display* root_display_ = nullptr;
```

Comment:
Consider making this pointer `const`

Reply:
Then I'll have to bother with `const_cast` since all functions that take it, require non-const pointer